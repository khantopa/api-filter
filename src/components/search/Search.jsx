import React from 'react';
import PropTypes from 'prop-types';
import './Search.scss';

const Search = (props) => {

  const handleChange = (e) => {
    props.onInputChange(e.target.value)
  }

  return (
    <div className="search">
      <label htmlFor={ props.label } className="search_label">{ props.label }:</label> 
      <input type="text" className="search_field" onChange={ handleChange } />
    </div>
  );
}

Search.propTypes = {
  onInputChange: PropTypes.func
}

export default Search;
