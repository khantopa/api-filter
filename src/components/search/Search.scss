@import '../../styles/main.scss';

.search {
  width: 100%;
  text-align: left;
  &_label {
    color: $dark;  
    margin-bottom: 40px;  
  }
  &_field {
    margin-top: 10px;
    @extend .form_field;
    text-align: left;
  }
}