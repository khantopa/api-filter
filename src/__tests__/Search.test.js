import Search from '../components/search/Search';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Search/> ); 
});

afterEach(() => {
  wrapper.unmount();
});

it('should render Search ', () => {
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
