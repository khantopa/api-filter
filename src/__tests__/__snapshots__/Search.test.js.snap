// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Search  1`] = `
<div
  className="search"
>
  <label
    className="search_label"
  >
    :
  </label>
  <input
    className="search_field"
    onChange={[Function]}
    type="text"
  />
</div>
`;
